src/utils.ts

```diff
@@ -456,3 +456,105 @@ export function setCrypto(c: Object) {
 export function getCrypto(): Object {
     return crypto;
 }
+
+// String averaging based upon https://stackoverflow.com/a/2510816
+// Dev note: We make the alphabet a string because it's easier to write syntactically
+// than arrays. Thankfully, strings implement the useful parts of the Array interface
+// anyhow.
+
+/**
+ * The default alphabet used by string averaging in this SDK. This matches
+ * all usefully printable ASCII characters (0x20-0x7E, inclusive).
+ */
+export const DEFAULT_ALPHABET = [""].reduce(() => {
+    let str = "";
+    for (let c = 0x20; c <= 0x7E; c++) {
+        str += String.fromCharCode(c);
+    }
+    return str;
+}, "");
+
+/**
+ * Pads a string using the given alphabet as a base. The returned string will be the
+ * same length as the alphabet, and padded with the first character in the alphabet.
+ *
+ * This is intended for use with string averaging.
+ * @param {string} s The string to pad.
+ * @param {string} alphabet The alphabet to use as a single string.
+ * @returns {string} The padded string.
+ */
+export function alphabetPad(s: string, alphabet = DEFAULT_ALPHABET): string {
+    return s.padStart(alphabet.length, alphabet[0]);
```

Comment:
padding at the start will cause the wrong behaviour for lexicographic sorting with mismatched string lengths, the sort will treat the strings differently to your `averageBetweenStrings` where `bb` ... `z` will be lexicographically sorted as such but `averageBetweenStrings` will treat `z` as ` z` which will be earlier than `bb`.

Reply:
Also padding to alphabet's length (94) is longer than MSC1772 allows (50) so this won't work for ordering space children.

Reply:
The padding is to remove floating point issues we have because JS doesn't trigger integer division the same way Python does, but I suppose we can throw in more `floor()`s.

Will take a look.

Also, 50 is a bit tiny if the character space is 95...

Reply:
Turns out I just padded the wrong value.